File: java/MobileExecute/src/main/java/configs/Configs.java
package configs;

import java.net.MalformedURLException;
import java.net.URL;

import org.openqa.selenium.remote.DesiredCapabilities;

public class Configs {

    static String USERNAME = "";
    static String ACCESS_KEY =  "";
    public static final URL kobitonServerUrl(){
        try {
            return new URL("https://" + USERNAME + ":" + ACCESS_KEY + "@api.kobiton.com/wd/hub");
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static final DesiredCapabilities desiredCapabilitiesAndroidApp(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "Java TestNG - Android App");
        capabilities.setCapability("sessionDescription", "This is an example for Android app testing");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/ApiDemos-debug.apk");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "*");
        capabilities.setCapability("platformName", "Android");
        capabilities.setCapability("platformVersion", "*");
        capabilities.setCapability("allowMobileShell", true);

        return capabilities;
    }
}

File: flutter/Ruby/configs.rb
USER_NAME = ""
API_KEY = ""

class DesiredCapabilities
 def desired_caps_android_app
  desired_caps = {
    :caps => {
      :sessionName =>        'Automation test session',
      :sessionDescription => '',
      :deviceOrientation =>  'portrait',
      :noReset =>            true,
      :fullReset =>          false,
      :captureScreenshots => true,
      :deviceGroup =>        'KOBITON',
      :deviceName =>         '*',
      :tagName =>            '',
      :platformName =>       'Android',
      :platformVersion =>    '*',
      :retry_backoff_time => 5000,
      :automationName => 'Flutter',
      :app => 'https://kobiton-devvn.s3-ap-southeast-1.amazonaws.com/apps-test/demo/app-debug-all.apk'
    },
    :appium_lib => {
      :server_url => "https://#{USER_NAME}:#{API_KEY}@api.kobiton.com/wd/hub",
      :wait_timeout => 20,
      :wait_interval => 1
    }.freeze
  }
   return desired_caps
 end

 def desired_caps_ios_app
   desired_caps = {
     :caps => {
       :sessionName =>        'iOS app',
       :sessionDescription => 'This is an example for iOS App testing', 
       :deviceOrientation =>  'portrait',  
       :captureScreenshots => true, 
       :deviceGroup =>        'KOBITON', 
       :deviceName =>         '*',
       :platformName =>       'iOS',
       :retry_backoff_time => 5000,
       :automationName => 'Flutter',
       :app => 'https://kobiton-devvn.s3-ap-southeast-1.amazonaws.com/apps-test/demo/Runner.zip'
     },
     :appium_lib => {
       :server_url => "https://#{USER_NAME}:#{API_KEY}@api-test.kobiton.com/wd/hub",
       :wait_timeout => 20,
       :wait_interval => 1
     }
   }.freeze
   return desired_caps
 end
end
